google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectSelector.java

```diff
@@ -400,18 +415,37 @@ private TreePath find(DefaultMutableTreeNode root, String str) {
       while (nodes.hasMoreElements()) {
         DefaultMutableTreeNode node = nodes.nextElement();
         if (node instanceof ResourceProjectModelItem
-            && str.equalsIgnoreCase(((ResourceProjectModelItem) node).getProjectId())) {
+            && str.equalsIgnoreCase(
+                ((ResourceProjectModelItem) node).getProject().getProjectId())) {
           return new TreePath(node.getPath());
         }
       }
       return null;
     }
 
-    public JComponent getInitialFocus() {
+    JComponent getInitialFocus() {
       return tree;
     }
 
-    public void initializeContent(String selectedProjectId) {
+    void setFilter(String filter) {
+      for (int i = 0; i < tree.getRowCount(); i++) {
+        TreePath path = tree.getPathForRow(i);
+        if (path.getLastPathComponent() instanceof GoogleUserModelItem) {
+          GoogleUserModelItem userModelItem = (GoogleUserModelItem) path.getLastPathComponent();
+          userModelItem.setFilter(filter);
+
+          boolean isExpanded = tree.isExpanded(path);
```

Comment:
nit: move this above where its used, or just inline tree#isExpanded

Reply:
Obsolete, PTAL.